src/chapters/az-cli/walkthrough_az-cli.rst

```diff
@@ -0,0 +1,525 @@
+
+The `JMESPath query<https://jmespath.org/>`_ value we will use is ``"[].urn"`` which means take the output list ``[]`` and instead of the complete image objects only output the value for each of their the ``urn`` properties. The result is a list of just URN values which is much easier to work with!
+
+.. sourcecode:: powershell
+
+    > az vm image list --query "[].urn"
+
+From here we can see the URN we need for the Ubuntu image is ``"Canonical:UbuntuServer:18.04-LTS:latest"``. Let's assign that value to a variable so we don't have to clutter our clipboard:
+
+.. sourcecode:: powershell
+    :caption: on Windows/PowerShell
+
+    > $ImageURN="Canonical:UbuntuServer:18.04-LTS:latest"
+
+.. sourcecode:: bash
+    :caption: on Linux/BASH
+
+    $ image_urn="Canonical:UbuntuServer:18.04-LTS:latest"
+
+Now we can reference the URN by its variable name ``$ImageURN`` or ``image_urn`` depending on your chosen shell.
+
+.. admonition:: tip
+
+    You can make use of a slightly more advanced query and in-line evaluation to do this in one step. Below we use a filter on the list to only output objects whose URN property ``contains`` the string Ubuntu. Then we pipe the filtered list and assign the first element's URN value to the variable.
+
+    .. sourcecode:: powershell
+        :caption: filtering the image list
+
+        > az vm image list --query "[? contains(urn, 'Ubuntu')] | [0].urn"
+
+    When we issue this command using in-line evaluation we can assign output directly to the variable:
+
+    .. sourcecode:: powershell
+        :caption: Windows/PowerShell
+
+        > $ImageURN="$(az vm image list --query "[? contains(urn, 'Ubuntu')] | [0].urn")" 
+
+    When using the BASH shell there is a known `issue <https://github.com/Azure/azure-cli/issues/8401>`_ with the default JSON format where it includes quote characters ``""`` around single string outputs. Unfortunately this can break commands and scripts in BASH so we need to request a TSV output format to correct it:
+
+    .. sourcecode:: bash
+        :caption: Linux/BASH
+
+        # -o: tsv sets the output to TSV format to remove the double quote characters
+        $ image_urn="$(az vm image list --query "[? contains(urn, 'Ubuntu')] | [0].urn" -o tsv)" 
+
+Provisioning
+------------
+
+Now that we have our image URN we can provision the VM. We will use the following values for the remaining arguments:
+
+- ``-n``: <name>-linux-vm
+- ``--size``: Standard_B2s
+- ``--admin-username``: student
+- ``--image``: the image URN [stored in a variable]
+
+.. note::
+
+    It is important that you use these exact values so that it is easier to help you if something goes wrong along the way.
+
+Let's create our VM! Note that this command will take some time to complete.
+
+.. sourcecode:: powershell
+    :caption: Windows/PowerShell
+
+    > az vm create -n <name>-linux-vm --size "Standard_B2s" --image "$ImageURN" --admin-username "student" --assign-identity
```

Comment:
first time running if students haven't created an RSA key they will get an error/warning: `az vm create -n paul-linux-vm --size "Standard_B2s" --image "Canonical:UbuntuServer:18.04-LTS:latest" --admin-username "student" --assign-identity
An RSA key file or key value must be supplied to SSH Key Value. You can use --generate-ssh-keys to let CLI generate one for you`

Reply:
another note if students get a weird Python error around not having the module 'antlr4' they can resolve it w/ `pip3 install antlr4-python3-runtime`